### DISCUSSION TITLE
How do I set the distance function the modl should use?

### DISCUSSION THREAD

**Discussion by PhilipMay** [embeddings-benchmark/mteb#199] [Q&A] [answered]
I have models that are trained for cosine sim. and others are trained for dot product.
How do I specify that for evaluation?

**Comment by Muennighoff** [comment]
For some tasks it is automatically computed with both e.g. here https://github.com/embeddings-benchmark/mteb/blob/2341c4892a0ef7b8baf5c9aa5906303d0b7e6839/mteb/evaluation/evaluators/PairClassificationEvaluator.py#L80 & here https://github.com/embeddings-benchmark/mteb/blob/2341c4892a0ef7b8baf5c9aa5906303d0b7e6839/mteb/evaluation/evaluators/SummarizationEvaluator.py#L110

I think we should do that for all other tasks where it's only cos sim right now (& maybe also make it configurable). It looks like it's Reranking & Retrieval where it's always cos sim right now 🤔 Do you want to open a PR to fix it?

**Comment by rasdani** [comment, marked as answer]
I found multiple locations, where one can set a similarity function. The one at the very bottom I tried out myself, but could not tell if it had any effect, since I was using normalized embeddings anyway.

You can choose the similarity function when initialising the Evaluator. The two in question would be `RetrievalEvaluator.py` and `RerankingEvaluator.py`.

https://github.com/embeddings-benchmark/mteb/blob/2341c4892a0ef7b8baf5c9aa5906303d0b7e6839/mteb/evaluation/evaluators/RetrievalEvaluator.py#L36

https://github.com/embeddings-benchmark/mteb/blob/2341c4892a0ef7b8baf5c9aa5906303d0b7e6839/mteb/evaluation/evaluators/RerankingEvaluator.py#L31

Or the one I tried was in `evaluate()` of `AbsTaskRetrieval.py`
https://github.com/embeddings-benchmark/mteb/blob/2341c4892a0ef7b8baf5c9aa5906303d0b7e6839/mteb/abstasks/AbsTaskRetrieval.py#L41

For this you would need to pass the kwarg here:
https://github.com/embeddings-benchmark/mteb/blob/2341c4892a0ef7b8baf5c9aa5906303d0b7e6839/mteb/evaluation/MTEB.py#L271

So basically in your run-Script (see examples), as `.run(..., score_function="dot")`, if I remember correctly.
